include/exceptions/io_exception.h

```diff
@@ -0,0 +1,17 @@
+#include "exception_base.h"
+#include <string>
+
+class CopyFileReadException : public CopyExceptionBase {
+    std::string _msg;
```

Comment:
Why do have another _msg member?
See, there is class CopyExceptionBase with _msg
You inherit from it and introduce another _msg member.
I don't see, why you invent such a member here.